File: my-folder/0495-teemo-attacking/solution.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int res = 0;
        int start = timeSeries[0];
        int end = timeSeries[0] + duration;
        for (int i = 1; i < timeSeries.length; i++) {
            if (timeSeries[i] > end) {
                res += end - start;
                start = timeSeries[i];
            }
            end = timeSeries[i] + duration;
        }
        res += end - start;
        return res;
    }
}

File: my-folder/0495-teemo-attacking/solution.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        res = 0
        start = timeSeries[0]
        end = timeSeries[0] + duration
        for i in range(1, len(timeSeries)):
            if timeSeries[i] > end:
                res += end - start
                start = timeSeries[i]
            end = timeSeries[i] + duration
        res += end - start
        return res
